src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -535,15 +557,33 @@ const StudentApplication = ({ userProfile, authentication }) => {
    * @function submitApplication
    */
   const submitApplication = async () => {
-    if (applicationDetails.Applicants.every((applicant) => isApplicantValid(applicant))) {
+    const showGenericSubmitError = () => {
+      setSnackbarText('Something went wrong while trying to submit the application.');
+      setSnackbarSeverity('error');
+      setSnackbarOpen(true);
+    };
+
+    if (!applicationDetails.Applicants.every((applicant) => isApplicantValid(applicant))) {
```

Comment:
If `isApplicantValid` is still an `async` function, then this won't behave as expected, because calling `isApplicantValid(applicant)` will return a `Promise`, which is true because it's an `object`.

Reply:
This is fixed in #1094 .